### DISCUSSION TITLE
Primeng and zoneless?

### DISCUSSION THREAD

**Discussion by BeGj** [primefaces/community#2310] [PrimeNG]
Hi.
Is primeng compatible with zonless change detection released in angular 18?

Are there works in progress for making it zoneless-compatible if not?

**Comment by AndreasHae** [comment]
Curious about this as well. Angular team recommends migrating components to [`ChangeDetectionStrategy.OnPush`](https://angular.dev/guide/experimental/zoneless#onpush-compatible-components). That means if we want to migrate our applications to Zoneless, primeng would need to support this as well because of change detection being cascading.

**Reply by BeGj** [reply]
I tried a few components without zone (onPush for component ofc) and for example the labels would not move out of input fields when input had data loaded programatically. It worked as normal with the new 18 zone, but not zoneless.
So it seems that currently it does not work perfectly out of the box.

Edit: a suggested change to primeng could be to migrate to using signals perhaps?

**Comment by vanuatoo** [comment]
I intend to write an application using Angular 18 or higher and PrimeNG and does not want to use Zone.js. I think PrimeNG should support Zoneless.

**Comment by faheemhameed** [comment]
I have started building an app with Angular 18 and PrimeNG v18. I am not sure that the latest version of PrimeNG does support zoneless. Any confirmation would be great!!

**Comment by cagataycivici** [comment]
Zoneless is an experimental feature, we'll work on it once it becomes production ready. We're updating the codebase with modern Angular APIs so definitely interested in removing zones to cleanup up our codebase as well.

**Reply by faheemhameed** [reply]
Many thanks!  I agree with you on zoneless. I can't wait to see your codebase updated with the modern Angular APIs. Great work indeed.

**Comment by nawlbergs** [comment]
:( i guess ill turn zoneless back off... since i want to use primeng.  :(

**Comment by magnolo** [comment]
i build some apps zoneless with angular 19 (without primeng) and i am very happy how it worked out so far. my apps are completly signal driven. have some other apps depending on primeng (because it is awesome). also really looking forward to go zoneless with primeng included in the future. thx for all your awesomeness ^^
